packages/apollo-federation/src/composition/validate/sdl/uniqueFieldDefinitionNames.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  ASTVisitor,
+  NameNode,
+  GraphQLError,
+  InputObjectTypeDefinitionNode,
+  InterfaceTypeDefinitionNode,
+  ObjectTypeDefinitionNode,
+  InputObjectTypeExtensionNode,
+  InterfaceTypeExtensionNode,
+  ObjectTypeExtensionNode,
+  GraphQLNamedType,
+  isObjectType,
+  isInterfaceType,
+  isInputObjectType,
+} from 'graphql';
+import { SDLValidationContext } from 'graphql/validation/ValidationContext';
```

Comment:
Nit:

I really enjoy using the path based imports in `graphql` since the package is pretty well organized and it makes it possible to understand at-a-glance where a particular symbol is actually exported from and what it's purpose is.  (Similarly, I dislike top-level entry points which export everything!).

That said, I find the mix of path-based (e.g. `from 'graphql/validation/ValidationContext'`, `from 'graphql/type/schema'`, etc.) and top-level `graphql` imports (e.g. `from 'graphql'`) to be a bit off-pattern (but certainly not _anti_!)  For example, I think all of the symbols above which end with `Node` could be imported directly from `graphql/language/ast` and those ending with `Type` could be brought in from `graphql/type/definition`?

Reply:
I'm neither for or against your preference, however the top-level import is a (symptom of the) simpler workflow via VSCode auto imports.

As I don't have much of an opinion, my only concern with adopting path-based is how to go about enforcing that style. I'd be happy to add somewhat of a style guide somewhere, but that still doesn't _really_ enforce it. Is something we could easily lint against?

Curious to hear your thoughts!

Reply:
If you gave someone a blank canvas here, they might go forward with using top-level imports and overlook and path based values (whether those be separations in logic or code-paths which could be tree-shaken off), but if the canvas is awash in path-based imports, their temptation might be to leverage that existing inspiration.

Enforcing it probably isn't simple.  And the practicality of using path-based imports is often dependent on external libraries who often consider it an implementation detail where they could move "internal" exports around freely while still exporting them from the same "main module".  While `graphql` takes great care to not do this, it's not always the case.

I think we could import everything from top-level `graphql` or use the path-based imports, but having both seems peculiar.  That said, let's not worry about this; it was just an observation.  (Also personally, I found turning off the auto-import feature of VSCode comforting since it often imported random unrequested modules that I would later need to remove.)

cc @trevor-scheer

Reply:
I'll leave as is then, per your recommendation, and keep in mind for future reviews and PRs.

Always appreciate hearing your perspective 😄